Tasks/HelmDeployV0/src/helm.ts

```diff
@@ -78,17 +78,18 @@ function runHelmSaveCommand(helmCli: helmcli, kubectlCli: kubernetescli, failOnS
 
 async function run() {
     var command = tl.getInput("command", true).toLowerCase();
+    var connectionType = tl.getInput("connectionType", true);
     var isKubConfigRequired = isKubConfigSetupRequired(command);
     var kubectlCli: kubernetescli;
-    if (isKubConfigRequired) {
-        var kubeconfigfilePath = command === "logout" ? tl.getVariable("KUBECONFIG") : await getKubeConfigFile();
+    var externalAuth = connectionType === "None" && tl.getVariable("KUBECONFIG");
```

Comment:
I'm thinking it's better to set externalAuth only if connectionType === "None" and command is "install" or "upgrade". If "KUBECONFIG" is not set, we can log an error directly indicting that KUBECONFIG must be set when command is "install" or "upgrade". That would be more dev-friendly and avoid calling kubectlCli.login() when not needed

Reply:
sounds good, updated